lib/mayaUsd/ufe/UsdTransform3dPointInstance.cpp

```diff
@@ -0,0 +1,270 @@
+        _scaleModifier.setPrimAndInstanceIndex(usdPrim, instanceIndex);
+
+        _positionModifier.setValue(_position, UsdTimeCode::Default());
+        _orientationModifier.setValue(_orientation, UsdTimeCode::Default());
+        _scaleModifier.setValue(_scale, UsdTimeCode::Default());
+    }
+
+private:
+    GfVec3f                             _position;
+    GfQuath                             _orientation;
+    GfVec3f                             _scale;
+    UsdPointInstancePositionModifier    _positionModifier;
+    UsdPointInstanceOrientationModifier _orientationModifier;
+    UsdPointInstanceScaleModifier       _scaleModifier;
+};
+
+} // anonymous namespace
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdTransform3dPointInstance::UsdTransform3dPointInstance(const UsdSceneItem::Ptr& item)
+    : UsdTransform3dBase(item)
+{
+    if (item != nullptr) {
+        const UsdPrim prim = item->prim();
+        const int     instanceIndex = item->instanceIndex();
+        _positionModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+        _orientationModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+        _scaleModifier.setPrimAndInstanceIndex(prim, instanceIndex);
+    }
+}
+
+/* static */
+UsdTransform3dPointInstance::Ptr UsdTransform3dPointInstance::create(const UsdSceneItem::Ptr& item)
+{
+    return std::make_shared<UsdTransform3dPointInstance>(item);
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::translation() const
+{
+    return _positionModifier.getUfeValue();
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::rotation() const
+{
+    return _orientationModifier.getUfeValue();
+}
+
+/* override */
+Ufe::Vector3d UsdTransform3dPointInstance::scale() const { return _scaleModifier.getUfeValue(); }
+
+/* override */
+Ufe::TranslateUndoableCommand::Ptr
+UsdTransform3dPointInstance::translateCmd(double x, double y, double z)
+{
+    return std::make_shared<UsdPointInstanceTranslateUndoableCommand>(
+#ifdef UFE_V2_FEATURES_AVAILABLE
+        path(),
+#else
+        usdSceneItem(),
+#endif
+        UsdTimeCode::Default());
+}
+
+/* override */
+Ufe::RotateUndoableCommand::Ptr UsdTransform3dPointInstance::rotateCmd(double x, double y, double z)
+{
+    return std::make_shared<UsdPointInstanceRotateUndoableCommand>(
+#ifdef UFE_V2_FEATURES_AVAILABLE
+        path(),
+#else
+        usdSceneItem(),
+#endif
+        UsdTimeCode::Default());
+}
+
+/* override */
+Ufe::ScaleUndoableCommand::Ptr UsdTransform3dPointInstance::scaleCmd(double x, double y, double z)
+{
+    return std::make_shared<UsdPointInstanceScaleUndoableCommand>(
+#ifdef UFE_V2_FEATURES_AVAILABLE
+        path(),
+#else
+        usdSceneItem(),
+#endif
+        UsdTimeCode::Default());
+}
+
+/* override */
+Ufe::SetMatrix4dUndoableCommand::Ptr
+UsdTransform3dPointInstance::setMatrixCmd(const Ufe::Matrix4d& m)
+{
+    return std::make_shared<UsdPointInstanceSetMatrix4dUndoableCmd>(path(), m);
```

Comment:
See above, where I question the need for this in practice.  If you feel there's a good use case for it, then you're good, otherwise I suggest some kind of TF_CODING_ERROR and returning nullptr.

Reply:
Doesn't sound like there is a good use case currently, so I'll do that, thanks!

Reply:
With the command class removed, this now issues a coding error and returns `nullptr` instead:
https://github.com/Autodesk/maya-usd/pull/1241/commits/ac913871efe412ad175fa1f1adfc2850b7ff56f2#diff-80faaef9cd24f227b2c4ada0deab537b01871c846af3c0cfa1aeb66d5b9dafd5R124